client/src/views/SharePoint.tsx

```diff
@@ -172,51 +182,84 @@ export const SharePoint = () => {
                             'Communications Sites Count': sitesCount.communicationSites,
                         }}
                     />
                 )}
-            </div>
-            <SharePointSitesList
-                handleRowClick={site => {
-                    setSelectedSite(site)
-                }}
-                height="14rem"
-                isLoading={isLoadingSiteActivities}
-                width="80%"
-                columnDefs={columnDefTopSites}
-                sites={topSitesByPageView}
-            />
-
-            <div style={{ display: 'flex', justifyContent: 'flex-end' }}>
-                <div>
-                    <AgChartsReact
-                        options={{ ...activityByGeoLocationOptions, width: 400, data: activityByGeoLocation }}
-                    />
-                </div>
-                <div>
-                    <AgChartsReact options={{ ...topSitesByPageViewOptions, width: 400, data: topSitesByPageView }} />
-                </div>
-            </div>
+            </Stack>
+            <Stack direction="row" alignItems="start" spacing={1}>
+                <SharePointSitesList
+                    handleRowClick={site => {
+                        setSelectedSite(site)
+                        setIsDrawerOpen(true)
+                    }}
+                    height="35rem"
+                    isLoading={isLoadingSiteActivities}
+                    width="75%"
+                    columnDefs={columnDefTopSites}
+                    sites={topSitesByPageView}
+                />
+                <Stack>
+                    <div>
+                        <AgChartsReact
+                            options={{ ...activityByGeoLocationOptions, width: 380, data: activityByGeoLocation }}
+                        />
+                    </div>
+                    <div>
+                        <AgChartsReact options={{ ...topSitesByPageViewOptions, width: 380, data: topSitesByPageView }} />
+                    </div>
+                </Stack>
+            </Stack>
+
+            <Stack direction="row">
+                <SharePointSitesList
+                    handleRowClick={site => {
+                        setSelectedSite(site)
+                        setIsDrawerOpen(true)
+                    }}
+                    height="25rem"
+                    isLoading={isLoadingSiteActivities}
+                    width="40%"
+                    columnDefs={columnDefSiteAudience}
+                    sites={[]}
+                />
+                <SharePointSitesList
+                    handleRowClick={site => {
+                        setSelectedSite(site)
+                        setIsDrawerOpen(true)
+                    }}
+                    height="25rem"
+                    isLoading={isLoadingSiteActivities}
+                    width="60%"
+                    columnDefs={columnDefSelectedSitePages}
+                    sites={selectedSitePages}
```

Comment:
This table data (sites) should be base on all sites available not just selected one
To get View (Unique and General):
You call
https://graph.microsoft.com/v1.0/sites/${SiteId}/analytics/allTime
e.g. https://graph.microsoft.com/v1.0/sites/aa46ece3-ff95-4665-8d26-8a8452c29627/analytics/allTime

In response:
You get and object with:
Actor Count = Unique Page viewers
Action Count = Page Views

Reference: MS graph API doc v2 - SharePoint

Reply:
Fixed now with p-limit and concurrency of 10